Document Scratch Storage self-management feature (#250)
nasa-fornax/fornax-documentation: documentation/data_storage.md

This storage is not backed up.
The storage system is highly reliable and should not lose data due to system error, but cannot protect against user error.
Lost data are not recoverable.

By default, `/scratch` exists but is limited to 1 GB.
You can expand it up to 3 TB for a maximum of 90 days at a time.
To manage this space, go to the [](#forsc-Dashboard) and click `Storage → Scratch`.
Details are in the drop-down below.

````{note} How to manage scratch storage space
:class: dropdown

On the [](#forsc-Dashboard), click `Storage → Scratch` to get to the Scratch Storage management page.

```{figure} ../_static/forsc_dashboard_scratch_default.png
:alt: Scratch Storage management page on the Fornax Dashboard, showing the default configuration (user has not created scratch space) and a Request Storage button.

Scratch Storage Dashboard page with a button to request storage
```

Create scratch space (expand beyond the default 1 GB)
: Go to the Scratch Storage Dashboard page and complete the following steps.

>   There is no limit on how many times you can do this, other than your remaining credits.

Reviewer: ```suggestion
  The only limit on how many times you can do this is the number of credits you have available.
```

  1. Click `Request Storage`.
  2. Choose a size between 500 GB and 3 TB.
     (You will be able to increase this later.)
  3. Choose an expiration date of up to 60 days in the future.
     (You will be able to increase or decrease this later.)
  4. The total charge in credits will be displayed, along with the number of credits you currently have.
     Be sure to leave yourself enough credits to run servers and anything else you want to do.
  5. Click `Submit Request`.
     If you have enough credits to cover the total, your scratch space will be expanded.
     This can take a few minutes.

>      When it's complete, you'll receive a message via the [Forum](#intro-forum), and your Dashboard page will look similar to the screenshot below.

Reviewer: I can confirm I got the message

```{figure} ../_static/forsc_dashboard_scratch_created.png
:alt: Scratch Storage management page on the Fornax Dashboard, showing the configuration of the user's current scratch space (storage size, created date, expiration date, and days remaining) and buttons to modify or release the storage space.

Scratch Storage Dashboard page after space has been created
```

Access scratch space
: [Start a server session](#start-server-session) and navigate to the `/scratch` directory.

>   Note that you won't be able to see `/scratch` in the UI because it doesn't show up in your home directory.

Reviewer: I can see it in the file finder on the left side of jupyterlab, it is in my home directory.

>   However, you can access it by (for example) opening a {term}`terminal` and entering `ls /scratch`.

Reviewer: ```suggestion
You can also access it by opening a {term}`terminal` and entering `ls /scratch`.
```

> Change scratch size or expiration date

Reviewer: ```suggestion
Change scratch size or expiration date.  Note this may take a little while (~20min)
```

Author: Did changing your scratch size or expiration date take awhile for you? For me those things were fairly quick. It was just deleting the storage that took awhile.

: Go to the Scratch Storage Dashboard page and click `Modify Allocation`.
  From there you will be able to:

  - Increase the storage size up to a maximum of 3 TB.
    (Can't decrease the size.)
  - Increase or decrease the expiration date.
    It can be increased to up to 90 days total from the date it was initially expanded.

Release (delete) scratch space
: Release the scratch space as soon as you're done with it so your credits stop being charged.
  Go to the Scratch Storage Dashboard page and click `Release Storage`.

>   You will be asked to confirm.

Reviewer: You will also get asked to close jupyterhub server to complete the release.  We should warn users that this may take a long time = 17 minutes for me just now.

  If you don't manually release the space, it will be automatically released on the expiration date you chose when creating or updating the space.

>   When the space is released, all files in `/scratch` will be **permanently deleted** and the directory will be resized back to 1 GB.

Reviewer: We should emphasize that this is really *all* files, not just those files you may have added since getting to the additional space, or not just those files above your standard allocation of 1GB, but all all all files, so be sure to save anything from scratch that you may want.   Maybe this should go into the danger warning below.  I first understood that warning to be able not recovering data, but also it should emphasize the "all" part.

Author: Updated. LMK if you think it still needs to be elaborated more.
